### ISSUE TITLE
Bug: Coverage not working with vitest workspace config

### ISSUE THREAD

**Issue by Shridhad** [cenfun/vitest-monocart-coverage#3]
We are using vitest workspace config with multiple packages in the monolith application. However, coverage collection fails in workspace mode. I tried to reproduce this issue with https://github.com/cenfun/merge-code-coverage-vitest and the same following error: 

```
[MCR] Not found coverage data in dir(s):
```

Full logs:
```shell
➜  merge-code-coverage-vitest git:(main) npm run test:unit

> merge-code-coverage-vitest@1.0.0 test:unit
> vitest run


 RUN  v2.0.5 /Users/shridhar/Documents/projects/personal/monocart/merge-code-coverage-vitest
      Coverage enabled with monocart

stdout | sum.test.js > sum 1 + 2 to equal 3
test sum in unit

 ✓ |unit| sum.test.js (1)
   ✓ sum 1 + 2 to equal 3

 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  16:56:10
   Duration  468ms (transform 14ms, setup 0ms, collect 12ms, tests 2ms, environment 0ms, prepare 256ms)

[MCR] Not found coverage data in dir(s):
➜  merge-code-coverage-vitest git:(main) 
```

Git repo: https://github.com/Shridhad/merge-code-coverage-vitest

## Steps:

1. Add vitest workspace and global config with workspace [coverage configuration](https://vitest.dev/guide/workspace.html#configuration)
2. Add package specific vitest config
3. Run `npm run test:unit`

## Code blocks: 

`./vitest.workspace.ts`: 
```ts
import { defineWorkspace } from "vitest/config";

// eslint-disable-next-line import/no-default-export
export default defineWorkspace(["test/unit/{vite,vitest}.config.*"]);
```

`./vitest.config.ts`:
```ts
import { defineConfig } from "vitest/config";
import coverageOptions from "./mcr.config.unit.js";

export default defineConfig({
  test: {
    // include: ['test/unit/*.test.js'],
    coverage: {
      enabled: true,
      include: ["src/**"],

      provider: "custom",
      customProviderModule: "vitest-monocart-coverage",

      // or a config path for coverage options
      // coverageReportOptions: "mcr.config.js"
      coverageReportOptions: coverageOptions,
    },
  },
});
```

`./packages/a/vitest.config.ts`:
```ts
import { defineProject } from "vitest/config";

const testConfig = defineProject({
  test: {
    include: ["./*.test.js"],
  },
});

export default testConfig;
```

**Comment by Shridhad** [comment]
Not sure if this vitest issue or an issue with the custom coverage provider(`MonocartProvider`). After trying to debug, I noticed, that when run with workspace config, the [onAfterSuiteRun](https://github.com/cenfun/vitest-monocart-coverage/blob/main/lib/index.js#L108) is not invoked at all.  As this is custom provider, I believe vitest invokes this overridden method. (or maybe I am wrong)

**Comment by cenfun** [comment]
I'm not familiar with `vitest.workspace`, could you please create a minimal reproducible repo for debug the issue?

**Comment by Shridhad** [comment]
Yes, I have it ready here. https://github.com/Shridhad/merge-code-coverage-vitest

**Comment by cenfun** [comment]
Yes, you are right. 
the `onAfterSuiteRun` is not invoked when run with workspace config, so `MCR` did not receive any coverage data. 
It should be a issue with Vitest.

**Comment by cenfun** [comment]
@Shridhad It should be fixed, please try vitest-monocart-coverage@2.1.1

test for https://github.com/Shridhad/merge-code-coverage-vitest
```sh
> npm run test:unit
> merge-code-coverage-vitest@1.0.0 test:unit
> vitest run


 RUN  v2.0.5 /bugs/merge-code-coverage-vitest
      Coverage enabled with monocart

stdout | sum.test.js > sum 1 + 2 to equal 3
test sum in unit

 ✓ |unit| sum.test.js (1)
 ✓ |unit| sum.test.js (1)
   ✓ sum 1 + 2 to equal 3

 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  08:50:24
   Duration  235ms (transform 10ms, setup 0ms, collect 11ms, tests 3ms, environment 0ms, prepare 76ms)

[MCR] Unit Coverage Report
┌────────────┬─────────┬────────────┬──────────┬───────────┬─────────┬─────────────────┐
│ Name       │   Bytes │ Statements │ Branches │ Functions │   Lines │ Uncovered Lines │
├────────────┼─────────┼────────────┼──────────┼───────────┼─────────┼─────────────────┤
│ src        │         │            │          │           │         │                 │
│ └ index.js │ 51.46 % │    35.71 % │   0.00 % │   25.00 % │ 55.17 % │ 12-28           │
├────────────┼─────────┼────────────┼──────────┼───────────┼─────────┼─────────────────┤
│ Summary    │ 51.46 % │    35.71 % │   0.00 % │   25.00 % │ 55.17 % │                 │
└────────────┴─────────┴────────────┴──────────┴───────────┴─────────┴─────────────────┘

```

**Comment by Shridhad** [comment]
@cenfun That works like a charm. Thank you so much for quick fix.
